addons/mail/models/ir_attachment.py

```diff
@@ -58,7 +58,9 @@ def register_as_main_attachment(self, force=True):
 
         if force or not related_record.message_main_attachment_id:
             with contextlib.suppress(AccessError):
-                related_record.message_main_attachment_id = self
+                not_allowed_mimetypes = ["application/xml", "application/pkcs7-mime"]
+                if self.mimetype not in not_allowed_mimetypes:
+                    related_record.message_main_attachment_id = self
```

Comment:
Hello, thank you for the PR!

Aren't there a million others not allowed?

Shouldn't we just list allowed mimetypes instead? I can only think about pdf and a few image formats that are allowed.

See `isImage` and `isPdf` in `FileModelMixin` (`.js` in master) used in `get attachmentsInWebClientView`.

Reply:
Localizations notably expect other formats than image / pdf. I would let people deal with their attachments as much as possible.

Reply:
What's the point? They are not going to be displayed if they don't fulfill the condition in the client code.

Then when using pager between the various attachments of a document (only those displayed, so pdf or image) it will set it as main attachment, regardless of what was the previous value.

Maybe the current code is just too naive, but I feel like the inconsistency on what is allowed will only lead to potential issues?

Considering how easy it is to change it without noticing in the web client, using `message_main_attachment_id` for business critical code looks like a bad idea.

Do people actually know they must stay on the correct attachment in the viewer to keep it as main attachment before doing any other action that would depend on it? And that if going previous/next of a non-client side supported attachment, they will never be able to set it back as main attachment from the UI?

Reply:
I don't know where you took your specs for "pdf or image only" but that was never part of the main attachment usage. We don't always use pdf or png in real life. Can't say much more, I am not in charge of the viewer part.

Reply:
The spec for the viewer is just the list of types that we can display natively in most browser and OS.

I think the current spec for the auto (re-)assignment of main attachment from viewer/pager is way too naive according to what you explain.

But if nobody complained yet I guess I'll just close my eyes on this PR, feel free to just make an ignore list.

Reply:
To be honest I don't know all use cases ... there is some custom code in accounting for example where they explicitly search for xml files as main attachment (I think linked to mandatory invoices sending to government).

All in all I would not change much this heuristic, it exists and voila. Adding exceptions will lead to other issues in the end.

Reply:
Hello,
@tde-banana-odoo @seb-odoo 
I have made a different PR https://github.com/odoo/odoo/pull/175459 for this fix in master for italian localization. Please have a look on it. 
Thank You!